packages/apollo-vault/src/execute_unlock.rs

```diff
@@ -76,6 +76,21 @@ where
         vault_token_amount: Uint128,
     ) -> Result<Response, ContractError> {
         let vault_token = self.base_vault.vault_token.load(deps.storage)?;
+        let vault_token_denom = vault_token.to_string();
+
+        // Check that exactly the right funds were sent
+        if info.funds.len() != 1
+            || info.funds[0].denom != vault_token_denom
+            || info.funds[0].amount != vault_token_amount
+        {
+            return Err(ContractError::UnexpectedFunds {
+                expected: vec![Coin {
+                    denom: vault_token.to_string(),
+                    amount: vault_token_amount,
+                }],
+                actual: info.funds.clone(),
+            });
+        }
```

Comment:
is this not handled by the vault_token.receive?

Reply:
It doesn't check if there are extra denoms in info.funds. I figured we might as well make it explicit that we only want exactly the right funds. Can refactor this into a helper function later on.